cpp/src/nvgraph_gdf.cu

```diff
@@ -515,6 +515,69 @@ gdf_error gdf_AnalyzeClustering_ratio_cut_nvgraph(gdf_graph* gdf_G,
 	return GDF_SUCCESS;
 }
 
+
+gdf_error gdf_extract_subgraph_vertex_nvgraph(gdf_graph* gdf_G,
+																							gdf_graph* result,
+																							gdf_column* vertices) {
+	GDF_REQUIRE(gdf_G != nullptr, GDF_INVALID_API_CALL);
+	GDF_REQUIRE((gdf_G->adjList != nullptr) || (gdf_G->edgeList != nullptr), GDF_INVALID_API_CALL);
+	GDF_REQUIRE(vertices != nullptr, GDF_INVALID_API_CALL);
+	GDF_REQUIRE(vertices->data != nullptr, GDF_INVALID_API_CALL);
+	GDF_REQUIRE(!vertices->valid, GDF_VALIDITY_UNSUPPORTED);
+
+	// Initialize Nvgraph and wrap the graph
+	nvgraphHandle_t nvg_handle = nullptr;
+	nvgraphGraphDescr_t nvgraph_G = nullptr;
+	NVG_TRY(nvgraphCreate(&nvg_handle));
+	GDF_TRY(gdf_createGraph_nvgraph(nvg_handle, gdf_G, &nvgraph_G, false));
+
+	// Create an Nvgraph graph descriptor for the result and initialize
+	nvgraphGraphDescr_t nv_result = nullptr;
+	NVG_TRY(nvgraphCreateGraphDescr(nvg_handle, &nv_result));
+
+	// Call Nvgraph function to get subgraph (into nv_result descriptor)
+	NVG_TRY(nvgraphExtractSubgraphByVertex(	nvg_handle,
+																					nvgraph_G,
+																					nv_result,
+																					(int*)vertices->data,
+																					vertices->size));
+
+	// Get the vertices and edges of the created subgraph to allocate memory:
+	nvgraphCSRTopology32I_st topo;
+	topo.source_offsets = nullptr;
+	topo.destination_indices = nullptr;
+	nvgraphTopologyType_t TT = NVGRAPH_CSR_32;
+	NVG_TRY(nvgraphGetGraphStructure(nvg_handle, nv_result, (void*)&topo, &TT));
+	if (TT != NVGRAPH_CSR_32)
+		return GDF_C_ERROR;
+	int num_verts = topo.nvertices;
+	int num_edges = topo.nedges;
+	result->adjList = new gdf_adj_list;
+	result->adjList->offsets = new gdf_column;
+	result->adjList->indices = new gdf_column;
+	result->adjList->ownership = 0;
```

Comment:
If ownership is set to 0, who will free the memory allocated in line 560 and 561?

Reply:
Output of the analytic is a graph.  This space is owned by the graph, the user will need to delete the graph when they are done with it.